### PR TITLE
Fix table alignment and padding.

### LINKED ISSUE #6228: Tests in "Implicit.Remoting.Tests.ps1" fails after #5568 was merged
Tests in "Implicit.Remoting.Tests.ps1" fails after #5568 was merged.

Here is the failure: https://ci.appveyor.com/project/PowerShell/powershell-f975h/build/v6.1.0-preview.587
```
  Describing Import-PSSession with FormatAndTypes
VERBOSE: Using Global AppVeyor Credential
    Context Importing format file works
      [-] Error occurred in Context block 640ms
        Expected strings to be the same, but they were different.
        Expected length: 48
        Actual length:   46
        Strings differ at index 37.
        Expected: '\r
        Width Height\r
        ----- ------\r
          123    456\r
        \r
        \r
        '
        But was:  '\r
        Width Height\r
        ----- ------\r
          123  456\r
        \r
        \r
        '
        ------------------------------------------------------^
        642:                 $originalLocalFormatting | Should Be $originalRemoteFormatting
        at Invoke-LegacyAssertion, C:\projects\powershell-f975h\src\powershell-win-core\bin\Release\netcoreapp2.0\win7-x64\publish\Modules\Pester\4.3.1\Functions\Assertions\Should.ps1: line 188
        at <ScriptBlock>, C:\projects\powershell-f975h\test\powershell\Modules\Microsoft.PowerShell.Utility\Implicit.Remoting.Tests.ps1: line 642
        at Invoke-Blocks, C:\projects\powershell-f975h\src\powershell-win-core\bin\Release\netcoreapp2.0\win7-x64\publish\Modules\Pester\4.3.1\Functions\SetupTeardown.ps1: line 140
        at Invoke-TestGroupSetupBlocks, C:\projects\powershell-f975h\src\powershell-win-core\bin\Release\netcoreapp2.0\win7-x64\publish\Modules\Pester\4.3.1\Functions\SetupTeardown.ps1: line 125
        at DescribeImpl, C:\projects\powershell-f975h\src\powershell-win-core\bin\Release\netcoreapp2.0\win7-x64\publish\Modules\Pester\4.3.1\Functions\Describe.ps1: line 157
```

### LINKED ISSUE #6234: FailFast: padding is not correct
Description
------------------
Running Start-DevPowerShell or running the executable "pwsh.exe" results in new PowerShell window, but once you enter any table command, the window crashes. 

Noticed that this error occurs with fixes for pull request #5568 
Works with earlier versions however.

https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs#L501

Assert Statement Fails

Steps to reproduce 
------------------
Run the v6.0.1 executable "pwsh.exe" from the 5.1 launcher or type Start-DevPowerShell
```powershell
Start-DevPowerShell
```
Then type in any table view command such as: 
```powershell
ls
```
Expected behavior
-----------------
Should run PowerShell v6.0.1 in a new window without crashes 

Actual behavior
---------------
Initially opens PowerShell v6.0.1, but when you type in any table view command, the window crashes. 

Environment data
----------------

When checking the version table in the original PowerShell Window:
```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      5.1.15063.909
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.15063.909
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```
After running the pwsh in one instance within the same window:
```powershell
> pwsh
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.0.1
PSEdition                      Core
GitCommitId                    v6.0.1
OS                             Microsoft Windows 10.0.15063
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

The original change #5568 to remove extra padding didn't take into account alignment.

Fix logic to accommodate left, center, and right alignment in table for only right side padding and added test.
Removed over active debug assert when formatting was necessary.

Marked `Set-Date` tests as `-Pending` due to #6227 

Fix https://github.com/PowerShell/PowerShell/issues/6228
Fix https://github.com/PowerShell/PowerShell/issues/6234

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by vors** [comment]
Does it fix the #6234 as well?

**Comment by sayujshajith** [comment]
This PR fixes a consistent crash that I was getting with any command that uses Format Table. 
The issue tracking that crash is here: https://github.com/PowerShell/PowerShell/issues/6234

**Comment by SteveL-MSFT** [comment]
I think I figured out why the tests are failing.  Working on a fix.

**Comment by SteveL-MSFT** [comment]
Linux/Mac failures due to https://github.com/PowerShell/PowerShell/issues/6227

**Comment by SteveL-MSFT** [comment]
Marked Set-Date tests requiring `sudo` as Pending for now due to #6227

**Review by daxian-dbw — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@SteveL-MSFT Can you add information about `Set-Date` change in the description.

**Comment by SteveL-MSFT** [comment]
@adityapatwardhan done

File: src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs [hunk 1]
```diff
@@ -497,8 +504,6 @@ private static string GenerateRowField(string val, int width, int alignment, Dis
             {
                 return s;
             }
-            // we have to pad
-            System.Diagnostics.Debug.Assert(finalValueDisplayLength == width - 1, "padding is not correct");
```
Comment (daxian-dbw) [inline src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs]: [This is not a blocking comment]
@SteveL-MSFT I just noticed that in the `switch (alignment)` block below, `addPadding` is checked only when padding for `left alignment`, not when padding for `center alignment`. This is not in the same pattern as in the `switch (alignment)` block in `if (currentValueDisplayLength < width)`. Just want to bring it up so you can take a another look to see if it's intended.

Comment (SteveL-MSFT) [reply src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs]: Good catch, let me check this.  I'll add an extra test for truncation.

**Review by daxian-dbw — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1 [hunk 2]
```diff
@@ -1,11 +1,11 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 Describe "Set-Date for admin" -Tag @('CI', 'RequireAdminOnWindows', 'RequireSudoOnUnix') {
-    It "Set-Date should be able to set the date in an elevated context" {
+    It "Set-Date should be able to set the date in an elevated context" -Pending {
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1]: Please don't mark these as pending... Pending on linux at most.

**Pull request merged**